lib/internal/util/inspect.js

```diff
@@ -1753,7 +1781,14 @@ function formatArray(ctx, value, recurseTimes) {
     if (!ObjectPrototypeHasOwnProperty(value, i)) {
       return formatSpecialArray(ctx, value, recurseTimes, len, output, i);
     }
-    ArrayPrototypePush(output, formatProperty(ctx, value, recurseTimes, i, kArrayType));
+    const parentName = getParentName(ctx);
```

Comment:
`parentValue` would only be used in case the `getSingularizeArrayValuesString` option would be set. Please only trigger that inside of the if statement.